engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -67,17 +67,110 @@ private class Factory(
             it.location
         }
         val furtherTypeRenameInstructions = typeRenameInstructions.values +
-                SharedTypesAnalysis(overallSchema, services, fieldInstructions, typeRenameInstructions)
-                    .getTypeRenames()
+            SharedTypesAnalysis(overallSchema, services, fieldInstructions, typeRenameInstructions)
+                .getTypeRenames()
+
+        val underlyingBlueprints = deriveUnderlyingBlueprints(furtherTypeRenameInstructions)
+        val underlyingTypeNameToOverallNameByService =
+            makeUnderlyingTypeNamesToOverallNameByService(services, underlyingBlueprints)
+        // the above feeds into the below
+        val overAllTypeNameToUnderlyingNameByService =
+            makeOverAllTypeNameToUnderlyingNameByService(
+                services,
+                underlyingBlueprints,
+                underlyingTypeNameToOverallNameByService
+            )
```

Comment:
Lets discuss this map approach in person and then potentially get rid of it

Reply:
Discussed with Franklin - I am leaving in the map even though we only use the key as a Set.  We can improve this later with more debugging on why this is not symmetric when used INSIDE the blueprint